crates/opencascade-sys/src/lib.rs

```diff
@@ -1079,9 +1096,12 @@ pub mod ffi {
         pub fn TopLoc_Location_Transformation(location: &TopLoc_Location) -> UniquePtr<gp_Trsf>;
 
         type Handle_Poly_Triangulation;
+
+        /// # Safety
+        /// This method is safe assuming that `triangulation` points to a valid constructed object or is a null pointer.
         #[cxx_name = "construct_unique"]
-        pub fn Handle_Poly_Triangulation_ctor(
-            triangulation: &Handle_Poly_Triangulation,
+        pub unsafe fn Handle_Poly_Triangulation_ctor(
+            triangulation: *const Poly_Triangulation,
```

Comment:
It was a mistake from my previous PR where I added the fn. TriangulationHandle -> TriangulationHanle isnt the most useful of fns. Triangulation -> TriangulationHanle makes actual sense to use.

Reply:
Can we find a way to make it not require `unsafe`? It's currently the only function marked as unsafe so it feels like there should be another way around it.

It would be helpful to add some example code so I can understand the usage of these functions more.

Reply:
Your point is true that it's unsafe either way. I just prefer consistency with the existing code so things can be refactored more easily (which is long overdue, but I haven't had a lot of time for this project recently).

Reply:
I think my preference is to write as little C++ as possible and keep things in Rust. So I lean towards the unsafe but real API. 

That said its not the hill I am going to die on, so Ill update this when I have some free time.

Reply:
Sorry, I know I can be pretty stubborn about arbitrary things. I'm also not 100% up to speed on undefined behavior in unsafe Rust vs. in C++. From what I understand, you can more easily run into cases of undefined behavior on the Rust side of things as the compiler will take more liberties compared to C (and maybe C++?). It might just be paranoia but I'd rather keep the unsafe parts as they are right now in the C++ boilerplate, until cxx.rs improves to reduce the need for that boilerplate.

Reply:
It can be depending on what you are doing. Here its fine assuming you follow the safety comment.

Lots of good unsafe knowledge can be found here if you are interested in diving deeper: https://doc.rust-lang.org/nomicon/

Reply:
The unsafe is just because we take a `*const` and using a pointer must be in an unsafe block.

Reply:
Moved to a reference fn that casts to a pointer internally.